src/Skia/Perspex.Skia/FormattedTextImpl.cs

```diff
@@ -75,36 +108,206 @@ public Size Measure()
 
         public void SetForegroundBrush(IBrush brush, int startIndex, int length)
         {
+            // Is this method even relevant now???
+            //
+            //var scb = brush as SolidColorBrush;
+            //if (scb != null)
+            //{
+            //	Layout.Attributes.Insert(brushAttr);
+            //}
         }
 
 
-        void Reload()
+        void Rebuild()
         {
+            var length = _text.Length;
+
+            _lines.Clear();
+
+            _skiaRects = new SKRect[length];
+            _skiaLines = new List<PerspexFormattedTextLine>();
+
+            int curOff = 0;
+            float curY = 0;
+
+            var metrics = Paint.FontMetrics;
+            var mTop = metrics.Top;  // The greatest distance above the baseline for any glyph (will be <= 0).
+            var mBottom = metrics.Bottom;  // The greatest distance below the baseline for any glyph (will be >= 0). 
+            var mLeading = metrics.Leading;  // The recommended distance to add between lines of text (will be >= 0). 
+
+            // This seems like the best measure of full vertical extent
+            float lineHeight = mBottom - mTop;
+
+            // Rendering is relative to baseline
+            LineOffset = -metrics.Top;
+
+            string subString;
+
+            byte[] bytes;
+            GCHandle pinnedArray;
+            IntPtr pointer;
+
+            for (int c = 0; curOff < length; c++)
+            {
+                float lineWidth = -1;
+                int measured;
+                int extraSkip = 0;
+                if (WidthConstraint <= 0)
+                {
+                    measured = length;
+                }
+                else
+                {
+                    float constraint = WidthConstraint;
+                    if (constraint > MAX_LINE_WIDTH)
+                        constraint = MAX_LINE_WIDTH;
+
+                    subString = _text.Substring(curOff);
+
+                    // TODO: This method is not linking into SkiaSharp so we must use the RAW buffer version
+                    //measured = (int)Paint.BreakText(subString, constraint, out lineWidth) / 2;
+                    bytes = Encoding.UTF8.GetBytes(subString);
+                    pinnedArray = GCHandle.Alloc(bytes, GCHandleType.Pinned);
+                    pointer = pinnedArray.AddrOfPinnedObject();
+                    // for some reason I have to pass nBytes * 2
+                    measured = (int)Paint.BreakText(pointer, (IntPtr)(bytes.Length * 2), constraint, out lineWidth) / 2;
+
+                    // some weird unicode byte issue again
+                    //if(subString.Length % 2 == 1)
+                    //{
+                    //	measured -= 1;
+                    //}
```

Comment:
As you can see in my first comment (TODO) the simpler BreakText api would be preferred but does not work yet in SkiaSharp, and so the lower level call was required. An older version of SkiaSharp had even some issues with that, but now that appears to be resolved. Can remove this comment.